src/main/java/com/sourcegraph/javagraph/SbtProject.java

```diff
@@ -0,0 +1,190 @@
+                if (project.equals(subproject)) {
+                    continue;
+                }
+                LOGGER.debug("found subproject: {}", subproject);
+                sourceUnits.addAll(processProject(path, subproject, reader, writer));
+            }
+            return sourceUnits;
+        }
+
+        // only a single project
+        SourceUnit unit = new SourceUnit();
+
+        unit.Name = subprojects.iterator().next();
+        unit.Type = "ScalaArtifact";
+
+        // parse dependencies
+        for (String dependency : executeSbtCommand("show libraryDependencies", reader, writer)) {
+            String prefix = "* ";
+            Preconditions.checkState(dependency.startsWith(prefix), "unexpected dependency prefix: %s", dependency);
+            dependency = dependency.substring(prefix.length());
+            if (dependency.length() == 0) {
+                continue;
+            }
+
+            String[] dep = dependency.split(":");
+            Preconditions.checkState(dep.length >= 3, "unexpected dependency format: %s", dependency);
+            RawDependency raw = new RawDependency(/*groupID=*/ dep[0],
+                    /*artifactID=*/ dep[1], /*version=*/ dep[2],
+                    /*scope=*/ dep.length >= 4 ? dep[3] : null, /*file=*/ null,
+                    PathUtil.relativizeCwd(path.toAbsolutePath()).toString());
+            raw.type = "jar";
+            unit.Dependencies.add(raw);
+        }
+
+        // parse source files
+        unit.Files = new ArrayList<>();
+        for (String sourceFile : executeSbtCommand("show unmanagedSources", reader, writer)) {
+            String prefix = "* ";
+            // ignore trailer: "[success] Total time: 0 s, completed Oct 19, 2017 2:16:57 PM"
+            if (sourceFile.startsWith(prefix)) {
+                unit.Files.add(sourceFile.substring(prefix.length()));
+            }
+        }
+
+        executeSbtCommand("show version", reader, writer);  // TODO: SourceUnit lacks a field for version
+
+        unit.Dir = path.getParent().toString();
+
+        Collection<SourceUnit> sourceUnits = new ArrayList<>();
+        sourceUnits.add(unit);
+        return sourceUnits;
+    }
+
+    private static Collection<String> executeSbtCommand(String command,
+            BufferedReader reader, BufferedWriter writer) throws IOException {
+        LOGGER.debug("sbt input: {}", command);
+        writer.write(command);
+        writer.newLine();
+        // emit a sentinel so we know when to stop reading
+        writer.write("eval \"EOF\"");
+        writer.newLine();
+        writer.flush();
+
+        Collection<String> lines = new ArrayList<>();
+        String line;
+        while ((line = reader.readLine()) != null) {
+            LOGGER.debug("sbt output: {}", line);
+            if (line.equals("[info] ans: String = EOF")) {
+                break;
+            }
+            String prefix = "[info] ";
+            if (!line.startsWith(prefix)) {
```

Comment:
we may want to keep this? I know we log certain warnings for  Maven

Reply:
I changed this to emit the error and warn messages via `LOGGER`.